test-automation-framework/org.wso2.carbon.automation.extensions/src/main/java/org/wso2/carbon/automation/extensions/servers/carbonserver/TestServerManager.java

```diff
@@ -95,23 +122,37 @@ public void configureServer() throws AutomationFrameworkException {
      * @throws java.io.IOException If an error occurs while copying the deployment artifacts into the
      *                             Carbon server
      */
-    public String startServer()
-            throws AutomationFrameworkException, IOException, XPathExpressionException {
-        if(carbonHome == null) {
+    public String startServer() throws AutomationFrameworkException, IOException, XPathExpressionException {
+        return startServer(StringUtils.EMPTY);
```

Comment:
It was wrong to throw implementation specific exceptions like XPathExpressionException. It was there in the original code, but we can fix it now since we are upgrading the version.